src/main/java/bio/terra/service/dataset/flight/LockDatasetBaseStep.java

```diff
@@ -0,0 +1,52 @@
+package bio.terra.service.dataset.flight;
+
+import bio.terra.service.dataset.Dataset;
+import bio.terra.service.dataset.DatasetDao;
+import bio.terra.service.dataset.exception.DatasetNotFoundException;
+import bio.terra.service.dataset.exception.InvalidLockUsageException;
+import bio.terra.service.job.JobMapKeys;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+
+import java.util.UUID;
+
+// I was very tempted to call this LockStep...
```

Comment:
I'm not understanding what this class adds. It looks like it is of no use to the UnlockStep. So it doesn't make sense to me to make it a base class.

Reply:
It is to share the logic for fetching a dataset name in the case where only a dataset id is available in the input params. Feels like it could also be on a Utils class but I want to avoid having too many of those. What do you think?